src/main/java/com/google/sps/app/servlets/GetTrip.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.app.servlets;
+
+import com.google.cloud.datastore.BaseEntity;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.sps.model.Trip;
+import com.google.sps.util.dataStoreHelper;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-trip")
+public class GetTrip extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String userID = request.getParameter("userID");
+        List<Trip> Trip = getTrip(userID);
+        Gson gson = new Gson();
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(Trip));
+    }
+
+    public List<Trip> getTrip(String userID) {
```

Comment:
Do you think a TreeSet ordered by date created or another order could be better than a list?

Reply:
I think List should be ok. Assuming that the trips are unique at this point. When creating the trip we should  probably ensure that they are unique.